### PR TITLE
gen-eastwest supports revision

### PR DESCRIPTION
**Pull request by stevenctl**
We should make it easy for users to deploy eastwest gateway for revisioned deployments. 
Should deal with https://github.com/istio/istio/issues/28647 and #27916

### CODE REVIEW

**Review by howardjohn — APPROVED** [review]

**Review by nmittler — APPROVED** [review]

File: samples/multicluster/gen-eastwest-gateway.sh [hunk 1]
```diff
@@ -17,6 +17,7 @@
 set -euo pipefail
 
 SINGLE_CLUSTER=0
+REVISION=""
```
Comment (linsun) [inline samples/multicluster/gen-eastwest-gateway.sh:20]: should we give "default" as the default value?

If so, revision is not strictly required.

**Comment by linsun** [comment]
LG, thank you @stevenctl for fixing this!  And thank you @irisdingbj for pointing out the issue!

File: samples/multicluster/gen-eastwest-gateway.sh (hunk 1 above)
Comment (stevenctl) [reply samples/multicluster/gen-eastwest-gateway.sh:20]: I'd rather have the zero value so that we don't have to keep defaulting in sync here.

**Comment by stevenctl** [comment]
Yes big thanks to @irisdingbj for pointing this out

File: samples/multicluster/gen-eastwest-gateway.sh (hunk 1 above)
Comment (linsun) [reply samples/multicluster/gen-eastwest-gateway.sh:20]: if it is 0 value, does it mean user will have to specify this param when installing eastwest gw using the .sh script?

**Pull request merged**